editor/scene_tree_editor.cpp

```diff
@@ -1353,8 +1353,8 @@ void SceneTreeDialog::_cancel() {
 
 void SceneTreeDialog::_select() {
 	if (tree->get_selected()) {
-		emit_signal(SNAME("selected"), tree->get_selected()->get_path());
 		hide();
+		emit_signal(SNAME("selected"), tree->get_selected()->get_path());
```

Comment:
I'd add an explanation so that we don't change it back by mistake:
```suggestion
		// The `selected` signal may cause another dialog to be displayed, so be sure to hide this one first.
		hide();
		emit_signal(SNAME("selected"), tree->get_selected()->get_path());
```